src/hamiltonian.jl

```diff
@@ -165,6 +177,13 @@ function neg_energy(
     return -dot(r, h.metric._temp) / 2
 end
 
+function neg_energy(
+    h::Hamiltonian{<:RankUpdateEuclideanMetric,<:GaussianKinetic}, r::T, θ::T
+) where {T<:AbstractVecOrMat}
+    M⁻¹ = h.metric.M⁻¹
+    return -r' * M⁻¹ * r / 2
```

Comment:
Note that because you've defined `M` as just the diagonal part, this equation is incorrect. The full one is https://github.com/mlcolab/Pathfinder.jl/blob/f4ca90dc3d91f077f479d13904a2b6bf99e8ee25/src/integration/advancedhmc.jl#L82, which uses https://github.com/mlcolab/Pathfinder.jl/blob/f4ca90dc3d91f077f479d13904a2b6bf99e8ee25/src/woodbury.jl#L384-L388